Source/JavaScriptCore/heap/IsoSubspace.cpp

```diff
@@ -93,28 +97,38 @@ void* IsoSubspace::tryAllocateFromLowerTier()
         return allocation->cell();
     };
 
-    if (!m_lowerTierFreeList.isEmpty()) {
-        PreciseAllocation* allocation = &*m_lowerTierFreeList.begin();
+    if (UNLIKELY(m_isPreciseOnly)) {
+        PreciseAllocation* allocation = PreciseAllocation::tryCreate(m_space.heap(), size, this, 0);
+        return allocation ? revive(allocation) : nullptr;
+    }
+
+    ASSERT_WITH_MESSAGE(cellSize() == size, "non-preciseOnly IsoSubspaces shouldn't have variable size");
+    if (!m_preciseTierFreeList.isEmpty()) {
+        PreciseAllocation* allocation = &*m_preciseTierFreeList.begin();
         allocation->remove();
         return revive(allocation);
     }
-    if (m_remainingLowerTierCellCount) {
-        PreciseAllocation* allocation = PreciseAllocation::tryCreateForLowerTier(m_space.heap(), cellSize(), this, --m_remainingLowerTierCellCount);
+    if (m_remainingPreciseTierCount) {
+        PreciseAllocation* allocation = PreciseAllocation::tryCreateForPreciseTier(m_space.heap(), size, this, --m_remainingPreciseTierCount);
         if (allocation)
             return revive(allocation);
     }
     return nullptr;
 }
 
-void IsoSubspace::sweepLowerTierCell(PreciseAllocation* preciseAllocation)
+void IsoSubspace::sweepPreciseTierCell(PreciseAllocation* preciseAllocation)
 {
-    preciseAllocation = preciseAllocation->reuseForLowerTier();
-    m_lowerTierFreeList.append(preciseAllocation);
+    if (UNLIKELY(m_isPreciseOnly)) {
+        preciseAllocation->destroy();
+        return;
+    }
```

Comment:
This should not be handled here, right? (Since precise-only is not precise-tier-cell).